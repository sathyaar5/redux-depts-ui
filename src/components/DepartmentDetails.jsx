import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Header from './Header';
import EmployeeDetails from './EmployeeDetails';

const DepartmentDetails = () => {
  const { deptName } = useParams();
  const departments = useSelector(state => state.departments);
  const [selectedDepartment, setSelectedDepartment] = useState(null);

  useEffect(() => {
    const storedDepartment = JSON.parse(localStorage.getItem('selectedDepartment'));
    if (storedDepartment && storedDepartment.deptName.toLowerCase().replace(/\s+/g, '-') === deptName.toLowerCase()) {
      setSelectedDepartment(storedDepartment);
    } else {
      const matchedDepartment = departments.find(
        department => department.deptName.toLowerCase().replace(/\s+/g, '-') === deptName.toLowerCase()
      );
      if (matchedDepartment) {
        setSelectedDepartment(matchedDepartment);
        localStorage.setItem('selectedDepartment', JSON.stringify(matchedDepartment));
      } else {
        setSelectedDepartment(null);
        localStorage.removeItem('selectedDepartment');
      }
    }
  }, [deptName, departments]);

  if (!selectedDepartment) {
    return;
  }

  const { employeeDetails, manager } = selectedDepartment;
  const numberOfEmployees = employeeDetails.length;

  return (
    <div>
      <Header
        department={selectedDepartment}
        numberOfEmployees={numberOfEmployees}
        manager={manager}
      />
      <EmployeeDetails employees={employeeDetails} />
    </div>
  );
};

export default DepartmentDetails;