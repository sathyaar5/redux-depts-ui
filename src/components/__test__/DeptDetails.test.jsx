import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import renderer from "react-test-renderer";

import DepartmentDetails from "../DepartmentDetails";
import store from "../../store";

test("renders department details correctly", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/technical-department"]}>
        <Routes>
          <Route path="/:deptName" element={<DepartmentDetails />} />
        </Routes>
      </MemoryRouter>
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
