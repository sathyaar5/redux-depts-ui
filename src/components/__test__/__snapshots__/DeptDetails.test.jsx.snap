// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders department details correctly 1`] = `null`;
